contracts/settlements/lib.rs

```diff
@@ -93,17 +90,14 @@ mod settlements {
         /// Creates a new contract.
         #[ink(constructor)]
         pub fn new(ticker1: Ticker, ticker2: Ticker) -> Self {
```

Comment:
Optional: If we remove the unnecessary parameter `&self` from `get_did` we can improve this to: 
```Rust
Self {
 ticker1, 
 ticker2,
 initialized: true, 
 identity: Self::get_did(Self::env().account_id()), 
 ...
}

Reply:
Yup.  I didn't like how that was.